### PR TITLE
DATAUP-314: test modal elimination III

### PR DESCRIPTION
**Pull request by ialarmedalien**
# Description of PR purpose/changes

More modals eliminated!

Integration tests timed out, but unit tests have passed.

# Jira Ticket / Issue #
e.g. https://kbase-jira.atlassian.net/browse/DATAUP-314
- [x] Added the Jira Ticket to the title of the PR (e.g. `DATAUP-69 Adds a PR template`)

# Testing Instructions
* Details for how to test the PR:
- [x] Tests pass locally and in GitHub Actions
- [x] Changes available by spinning up a local narrative and seeing that nothing has changed. How boring!

# Dev Checklist:

- [x] My code follows the guidelines at https://sites.google.com/lbl.gov/trussresources/home?authuser=0
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] (JavaScript) I have run Prettier and ESLint on changed code manually or with a git precommit hook

### CODE REVIEW

File: test/unit/spec/narrative_core/jobCommChannel-spec.js [hunk 1]
```diff
@@ -105,28 +106,70 @@ define(['jobCommChannel', 'base/js/namespace', 'common/runtime'], (
...
-            ['request-job-update', { jobId: 'someJob', parentJobId: 'someParent' }],
-            ['request-job-completion', { jobId: 'someJob' }],
-            ['request-job-log', { jobId: 'someJob', options: {} }],
-            ['request-latest-job-log', { jobId: 'someJob', options: {} }],
-            ['request-job-info', { jobId: 'someJob', parentJobId: 'someParent' }],
+            { channel: 'ping-comm-channel', message: { pingId: 'ping!' }, type: 'ping' },
+            {
+                channel: 'request-job-cancellation',
+                message: { jobId: 'someJob' },
+                type: 'cancel_job',
```
Comment (ialarmedalien) [inline test/unit/spec/narrative_core/jobCommChannel-spec.js:113]: add in some extra info to test

File: test/unit/spec/narrative_core/jobCommChannel-spec.js [hunk 2]
```diff
@@ -399,26 +443,21 @@ define(['jobCommChannel', 'base/js/namespace', 'common/runtime'], (
...
+                        expect(document.querySelector(modalQuerySelector)).not.toBeNull();
                         // click the 'OK' button
                         document
                             .querySelector('.modal a.btn.btn-default.kb-job-err-dialog__button')
                             .click();
                         // expect it to be gone
-                        return new Promise((resolve) => {
-                            setTimeout(() => resolve(), 500);
+                        return TestUtil.waitForElementState(document.body, () => {
+                            return document.querySelectorAll('.modal').length === 0;
```
Comment (ialarmedalien) [inline test/unit/spec/narrative_core/jobCommChannel-spec.js:456]: i.e. the modal has disappeared

File: test/unit/spec/nbextensions/bulkImportCell/main-spec.js [hunk 3]
```diff
@@ -19,82 +19,101 @@ define([
...
-            spyOn(Jupyter.notebook, 'get_cells').and.callThrough();
-            return Main.load_ipython_extension().then(() => {
-                expect(Jupyter.notebook.get_cells).toHaveBeenCalled();
+            it('should initialize the notebook', () => {
+                // expect that setupNotebook was called, which we can
+                // proxy by checking on Jupyter.notebook.get_cells.
+                spyOn(Jupyter.notebook, 'get_cells').and.callThrough();
+                spyOn(BulkImportCell, 'make').and.callFake(() => {
+                    // fake out the call to 'make' the bulk import cell
+                    // as the mock does not have all the metadata
```
Comment (ialarmedalien) [inline test/unit/spec/nbextensions/bulkImportCell/main-spec.js:47]: that creates a error dialog, so faking out the function means the error never gets called

File: test/unit/spec/nbextensions/bulkImportCell/main-spec.js [hunk 4]
```diff
@@ -19,82 +19,101 @@ define([
...
-                    type: 'code',
-                    index: 1,
-                    cell: newCell,
-                    data: {
-                        type: 'app-bulk-import',
-                        typesToFiles: {
-                            fileType: {
-                                files: ['a_file'],
-                                appId: 'someApp',
+        describe('a newly-inserted code cell', () => {
```
Comment (ialarmedalien) [inline test/unit/spec/nbextensions/bulkImportCell/main-spec.js:56]: slight reorganisation here

File: test/unit/spec/narrative_core/jobCommChannel-spec.js [hunk 5]
```diff
@@ -105,28 +106,70 @@ define(['jobCommChannel', 'base/js/namespace', 'common/runtime'], (
...
                 return comm
                     .initCommChannel()
                     .then(() => {
                         expect(comm.comm).not.toBeNull();
-                        spyOn(comm.comm, 'send');
-                        testBus.emit(testCase[0], testCase[1]);
-                        return new Promise((resolve) => setTimeout(resolve, 4000));
+                        return new Promise((resolve) => {
+                            // resolve the promise when we see comm.comm.send(...)
+                            spyOn(comm.comm, 'send').and.callFake((...args) => {
```
Comment (briehl) [inline test/unit/spec/narrative_core/jobCommChannel-spec.js:155]: Much better way of handling this!

**Review by briehl — APPROVED** [review]
Seems fine, mostly linting and reorganizing, with a clever rewrite of the jobCommChannel stuff.

**Pull request merged**